runtime/jvmti/jvmtiStackFrame.c

```diff
@@ -415,14 +416,27 @@ jvmtiGetFrameLocation(jvmtiEnv* env,
 		rc = getVMThread(currentThread, thread, &targetThread, TRUE, TRUE);
 		if (rc == JVMTI_ERROR_NONE) {
 			J9StackWalkState walkState;
-
-			vm->internalVMFunctions->haltThreadForInspection(currentThread, targetThread);
-
-			walkState.walkThread = targetThread;
 			walkState.flags = J9_STACKWALK_INCLUDE_NATIVES | J9_STACKWALK_VISIBLE_ONLY | J9_STACKWALK_COUNT_SPECIFIED | J9_STACKWALK_RECORD_BYTECODE_PC_OFFSET;
 			walkState.skipCount = (UDATA) depth;
 			walkState.maxFrames = 1;
-			vm->walkStackFrames(currentThread, &walkState);
+
+#if JAVA_SPEC_VERSION >= 19
+			if (NULL == targetThread) {
+				j9object_t threadObject = J9_JNI_UNWRAP_REFERENCE(thread);
+				if (IS_VIRTUAL_THREAD(currentThread, threadObject)) {
+					j9object_t contObject = (j9object_t)J9VMJAVALANGVIRTUALTHREAD_CONT(currentThread, threadObject);
+					vmFuncs->walkContinuationStackFrames(currentThread, contObject, &walkState);
+				} else {
+					Assert_JVMTI_unreachable();
+				}
```

Comment:
To make it more clear that `NULL == targetThread` is only possible for a virtual thread, it can be rewritten as:
```suggestion
				/* targetThread can only be NULL for a virtual thread. */
				Assert_JVMTI_true(IS_VIRTUAL_THREAD(currentThread, threadObject));
				j9object_t contObject = (j9object_t)J9VMJAVALANGVIRTUALTHREAD_CONT(currentThread, threadObject);
				vmFuncs->walkContinuationStackFrames(currentThread, contObject, &walkState);

```

Reply:
Ignore this suggestion. Asserting in the main path all the time will be bad for perf.